src/averageAges.js

```diff
@@ -47,15 +60,30 @@ function calculateWomenAverageAge(people, withChildren) {
  * mother in the array. (A mother's age at child birth)
  *
  * If `onlyWithSon` is specified then function calculates age difference only
- * for sons and their mothers.
+ * for Moms and their mothers.
  *
  * @param {object[]} people
  * @param {boolean} onlyWithSon - optional
  *
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const findMotherAndSon = child => people
+    .find(mother => child.mother === mother.name)
+      && child.sex === 'm';
+
+  const findMotherAndChilds = child => people
+    .find(mother => child.mother === mother.name);
+
+  const childs = !onlyWithSon
+    ? people.filter(findMotherAndChilds)
+    : people.filter(findMotherAndSon);
+
+  const childsAndMomsDifference = (total, child) =>
+    total + (child.born - people.find(mother =>
+      child.mother === mother.name).born);
```

Comment:
Will be easier if you create variable for mother

```suggestion
  const childsAndMomsDifference = ((total, child) => {
    variableWithMother;

    return total + (child.born - variableWithMother.born;
  });
```